gpii/node_modules/flowManager/test/SettingsDataSourceTests.js

```diff
@@ -0,0 +1,280 @@
+                .reply(eachConfig.status, eachConfig.response);
+        } else if (index === "untrustedSettings") {
+            // mock GET requests to "/untrusted-settings"
+            cloudMock.get(eachConfig.url)
+                .matchHeader("Authorization", /Bearer .*/ )
+                .reply(eachConfig.status, eachConfig.response);
+        }
+    });
+};
+
+// The customized settingsDataSource component for tests
+fluid.defaults("gpii.tests.settingsDataSource", {
+    gradeNames: ["gpii.flowManager.untrusted.settingsDataSource"],
+    untrustedSettingsUrl: gpii.tests.settingsDataSource.hostname + gpii.tests.settingsDataSource.untrustedSettingsUrl,
+    accessTokenUrl: gpii.tests.settingsDataSource.hostname + gpii.tests.settingsDataSource.accessTokenUrl,
+    clientCredentialFilePath: "%universal/gpii/node_modules/flowManager/test/data/clientCredential-correct.json"
+});
+
+// The base testEnvironment grade to be inherited by all tests
+fluid.defaults("gpii.tests.settingsDataSourceTests", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    settingsDataSourceGrade: "gpii.tests.settingsDataSource",   // supplied by individual tests
+    testCaseHolderGrade: null,   // supplied by individual tests
+    distributeOptions: {
+        settingsDataSourceGrade: {
+            source: "{that}.options.settingsDataSourceGrade",
+            target: "{that > settingsDataSource}.type"
+        },
+        testCaseHolderGrade: {
+            source: "{that}.options.testCaseHolderGrade",
+            target: "{that > testCaseHolder}.type"
+        }
+    },
+    components: {
+        settingsDataSource: {
+            type: "gpii.tests.settingsDataSource"
+        },
+        testCaseHolder: {
+            type: "fluid.test.testCaseHolder"
+        }
+    }
+});
+
+// The base testCaseHolder grade to be inherited by test case holders used by all tests
+fluid.defaults("gpii.tests.settingsDataSource.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    events: {
+        onResponse: null,
+        onError: null
+    }
+});
+
+// 1. No saved access token: retrieve and save an access token that then is used to retrieve user settings
+fluid.defaults("gpii.tests.settingsDataSource.testCaseHolder.noSavedAccessToken", {
+    gradeNames: "gpii.tests.settingsDataSource.testCaseHolder",
+    modules: [{
+        name: "A workflow with undefined initial access token: an access token is requested, saved and used to retrieve user settings",
+        expect: 4,
+        tests: [{
+            name: "Settings received with the initial new token being fetched and saved",
+            sequence: [{
+                funcName: "gpii.tests.settingsDataSource.setupNock",
+                args: [gpii.tests.settingsDataSource.testCases.nockConfig]
+            }, {
+                func: "jqUnit.assertUndefined",
+                args: ["No saved access token", "{settingsDataSource}.accessToken"]
```

Comment:
The path that is being tested here is different from the one that we test below for success (`"{settingsDataSource}.gpii-token-test.accessToken"`).